clusterloader2/pkg/measurement/common/slos/pod_startup_latency.go

```diff
@@ -131,31 +129,23 @@ func (p *podStartupLatencyMeasurement) start(c clientset.Interface) error {
 	klog.Infof("%s: starting pod startup latency measurement...", p)
 	p.isRunning = true
 	p.stopCh = make(chan struct{})
-	p.informer = informer.NewInformer(
+	i := informer.NewInformer(
 		c,
 		"pods",
 		p.namespace,
 		p.fieldSelector,
 		p.labelSelector,
 		p.checkPod,
 	)
-
-	go p.informer.Run(p.stopCh)
-	timeoutCh := make(chan struct{})
-	timeoutTimer := time.AfterFunc(informerSyncTimeout, func() {
-		close(timeoutCh)
-	})
-	defer timeoutTimer.Stop()
-	if !cache.WaitForCacheSync(timeoutCh, p.informer.HasSynced) {
-		return fmt.Errorf("timed out waiting for caches to sync")
-	}
-	return nil
+	return informer.StartAndSync(i, p.stopCh, informerSyncTimeout)
 }
 
 func (p *podStartupLatencyMeasurement) stop() {
 	if p.isRunning {
 		p.isRunning = false
-		close(p.stopCh)
+		if p.stopCh != nil {
```

Comment:
p.isRunning is set together with p.stopCh.

The measurement is not thread-safe anyway, so I don't think this if is needed for anything.

Reply:
Done.